bin/oneoff/backfill_data/backfill_script_announcement_keys.rb

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
+
+# In order to unqiuely identify the announcements associated
+# with a Script, we are generating a UUID key. This can then
+# be used to accurately look up the translation amongst a
+# Script's array of announcements.
+
+require_relative '../../../dashboard/config/environment'
+
+def backfill_script_announcement_keys
+  Script.all.each do |script|
+    next unless script.announcements
+
+    # Create UUID key for each announcement
+    script.announcements.each do |announcement|
+      announcement[:key] = SecureRandom.uuid unless announcement[:key]
+    end
+    begin
+      script.save
+    rescue Exception => err
+      puts "Skipping #{script.id} - #{script.name} because of error:"
+      puts err.message
+      next
+    end
+
+    # Update its script_json
+    script.write_script_json
```

Comment:
Double checking here that this is the right approach in my attempt to update the source of truth(s)?

Reply:
We generally do not update script_json files in production. I think best practice is to update them in levelbuilder and then have the change propagate to all the over environments from there.  So maybe this oneoff script should be run in levelbuilder instead of production. 

@davidsbailey I think it would be good to have your input here as well

Reply:
@dmcavoy correct. my suggestion would be to add a protection to this script so that it errors out if run outside of levelbuilder or development, since there is otherwise the assumption that backfill scripts are meant to be run in production

Reply:
Sounds good! I will add a conditional check for the levelbuilder environment